evadb/third_party/vector_stores/weaviate.py

```diff
@@ -0,0 +1,173 @@
+# coding=utf-8
+# Copyright 2018-2023 EvaDB
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+from typing import List
+
+from evadb.third_party.vector_stores.types import (
+    VectorIndexQuery,
+    VectorStore,
+)
+from evadb.utils.generic_utils import try_to_import_weaviate_client
+
+required_params = []
+_weaviate_init_done = False
+
+
+class WeaviateVectorStore(VectorStore):
+    def __init__(self, **kwargs) -> None:
+        try_to_import_weaviate_client()
+        global _weaviate_init_done
+
+        # Get the API key.
+        self._api_key = kwargs.get("WEAVIATE_API_KEY")
+
+        if not self._api_key:
+            self._api_key = os.environ.get("WEAVIATE_API_KEY")
+
+        assert (
+            self._api_key
+        ), "Please set your Weaviate API key in evadb.yml file (third_party, weaviate_api_key) or " \
```

Comment:
There is no evadb.yml anymore. Could you update the message? Thanks.

Reply:
Thanks Andy! I've reorganized my code structure for the weaviate integration and added the relevant documentation and tests.